lib/spack/spack/test/util/binary_resource.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+import shutil
+
+import pytest
+
+import spack.util.binary_resource as br
+import spack.util.executable
+from spack.paths import spack_root
+
+datadir = os.path.join(spack_root, "lib", "spack", "spack", "test", "data", "binary_resource")
+file_file = os.path.join(datadir, "file.bat")
+
+pytestmark = pytest.mark.only_windows("only runs on windows")
+
+
+@pytest.fixture
+def mock_binary_resource_root(monkeypatch, tmpdir):
+    resource_root = tmpdir.mkdir("binary-resources")
+
+    def _binary_resource_root():
+        return resource_root
+
+    monkeypatch.setattr(spack.util.binary_resource, "binary_resource_root", _binary_resource_root)
+
+
+@pytest.fixture
+def no_system(working_env):
+    """Fixture preventing system binaries from being detected"""
+    os.environ["PATH"] = ""
+
+
+@pytest.fixture
+def tar_dir():
+    return os.path.dirname(spack.util.executable.which("tar").path)
```

Comment:
I think it would be more-straightforward to name this fixture `add_tar_to_path` (IMO it could also handle modifying/appending to `PATH` - it can take `no_system` as a param to enforce fixture ordering): it was initially confusing what this was for until I read through how it was used.